apps/src/music/views/MusicView.jsx

```diff
@@ -220,6 +221,12 @@ class UnconnectedMusicView extends React.Component {
   }
 
   async onLevelLoad(levelData, initialSources) {
+    const blockMode =
+      initialSources?.labConfig?.music?.blockMode ||
+      levelData?.blockMode ||
+      BlockMode.SIMPLE2;
```

Comment:
let's create a selector for this, so we can consolidate the priority logic and reuse across the different views? something like (in musicRedux)

```
export const getBlockMode = (state: RootState): ValueOf<typeof BlockMode> => {
  const {initialSources, levelProperties} = state.lab;
  return (
    (initialSources?.labConfig?.music.blockMode as ValueOf<typeof BlockMode>) ||
    (levelProperties?.levelData as MusicLevelData | undefined)?.blockMode ||
    BlockMode.SIMPLE2
  );
};
```

then in `MusicView`, we can pass it through as a prop in the `connect` function like
```
const MusicView = connect(
  state => ({
    ...
    blockMode = getBlockMode(state),
    ...
  }),
  ... 
)
```
where it can be accessed as a prop (`this.props.blockMode`)

And then in `Timeline`, we can simply use the selector like
```
const blockMode = useSelector(getBlockMode);
```

Reply:
done here: https://github.com/code-dot-org/code-dot-org/pull/61158